client/src/main/scala/com/dwolla/cloudflare/domain/model/ratelimits/Implicits.scala

```diff
@@ -17,11 +17,15 @@ private[cloudflare] object Implicits {
   }
 
   implicit def toModel(dto: RateLimitActionDTO): RateLimitAction = {
-    RateLimitAction(
-      mode = dto.mode,
-      timeout = dto.timeout,
-      response = dto.response.map(toModel)
-    )
+    dto.mode match {
```

Comment:
I wonder if the DTO here should match the ADT structure we're using the in model? We may need to provide some [extra help to circe for serialization and deserialization](https://circe.github.io/circe/codecs/custom-codecs.html) but it would prevent someone using only the DTO library from making the same mistakes we're trying to prevent by introducing the ADT into the model.

Decoder might look something like this?

```scala
  implicit val rateLimitActionDtoDecoder = new Decoder[RateLimitActionDto] {
    override def apply(c: HCursor): Decoder.Result[RateLimitActionDto] =
      for {
        mode ← c.downField("mode").as[String]
        timeout ← c.downField("timeout").as[Option[Int]]
        response ← c.downField("response").as[Option[String]]
      } yield (mode, timeout) match {
        case ("challenge", None) ⇒ ChallengeRateLimitActionDto
        case ("js_challenge", None) ⇒ JsChallengeRateLimitActionDto
        case (_, Some(t)) ⇒ TimeoutRateLimitActionDto(mode, t, response)
      }
  }
```